Sources/BraveWallet/Settings/Web3SettingsView.swift

```diff
@@ -171,6 +171,17 @@ private struct WalletSettingsView: View {
   }
 
   var body: some View {
+    if !keyringStore.isDefaultKeyringCreated {
+      // `KeyringStore` is optional in `Web3SettingsView`, but observed here.
+      // When wallet is reset, we need SwiftUI to be notified `isDefaultKeyringCreated`
+      // changed so we can hide Wallet specific sections
+      EmptyView()
+    } else {
+      sections
+    }
```

Comment:
You can technically. just do 

```swift
if keyringStore.isDefaultKeyringCreated {
  sections
}
```

The ViewBuilder will implicitly make an `EmptyView` for the else state

Reply:
Good point. I think I will leave it as-is for now so we can include the comment and don't mistakenly keep this section around in the future 🙂.